chore: Adds some details to the `Key Attestation` (#375)
italia/eidas-it-wallet-docs: docs/en/wallet-attestation.rst

.. note::
  It is not necessary to send the Wallet Hardware public key because it is already included in the ``key_attestation``.

>   As we have seen in the previous steps in fact, the Device Integrity Service (DIS) creates a Key Attestation linked
>   to the provided "challenge" and the public key of the Wallet Hardware. This process eliminates the need to send the
>   Wallet Hardware public key directly, as it is already included in the key attestation.
>   The ``hardware_key_tag`` serves as a reference or identifier for the corresponding Cryptographic Hardware key stored by
>   the Wallet Provider. Therefore, the Wallet Provider can associate the received ``hardware_key_tag`` with the appropriate
>   Cryptographic Hardware key in its storage.

Reviewer: ```suggestion
  As seen in the previous steps, the Device Integrity Service (DIS) creates a Key Attestation linked to the provided "challenge" and the public key of the Wallet Hardware. This process eliminates the need to send the Wallet Hardware public key directly, as it is already included in the key attestation. The ``hardware_key_tag`` serves as a reference or identifier for the corresponding Cryptographic Hardware key stored by the Wallet Provider. Therefore, the Wallet Provider can associate the received ``hardware_key_tag`` with the appropriate Cryptographic Hardware key in its storage.
```

    3. It MUST verify that the ``challenge`` was generated by  Wallet Provider and has not already been used.
    4. It MUST check that there is a Wallet Instance registered with that ``hardware_key_tag`` and that it is still valid.
    5. It MUST reconstruct the ``client_data`` via the ``challenge`` and the ``jwk`` public key, to validate ``hardware_signature`` via the Cryptographic Hardware Key public key registered and associated with the Wallet Instance.

>     6. It MUST validate the ``integrity_assertion`` as defined by the device manufacturers' guidelines. This check summarized here includes many checks that the wallet provider will have to perform and that are recommended by the operating system manufacturers. The formal checks to be performed can be found in the relevant documentation.

Reviewer: ```suggestion
    6. It MUST validate the ``integrity_assertion`` as defined by the device manufacturers' guidelines. The list of checks that the Wallet Provider MUST perform are defined by the operating system manufacturers documentation.
```

    "exp": 1687288395
  }

> **Step 18**: The response is returned by the Wallet Provider. If successful, the HTTP response code MUST be set to 200 OK and contain the Wallet Attestation signed by the Wallet Provider. The Wallet Instance therefore performs security and integrity verification about the Wallet Attestation.

Reviewer: ```suggestion
**Step 18**: The response is returned by the Wallet Provider. If successful, the HTTP response code MUST be set with the value ``200 OK`` and contain the Wallet Attestation signed by the Wallet Provider. The Wallet Instance therefore performs security, integrity and trust verification about the Wallet Attestation and its issuer.
```
